### PR TITLE
feat(bazed-svelte): Display selections in the frontend

### PR DESCRIPTION
**Pull request by druskus20**

### CODE REVIEW

File: node_packages/bazed-svelte/src/lib/Rekuho.svelte [hunk 1]
```diff
@@ -77,7 +80,32 @@
...
+    ) {
+      start = caret.tail
+      end = caret.head
+    }
+
+    return {
+      start,
+      end,
+    }
+  }
```
Comment (elkowar) [inline node_packages/bazed-svelte/src/lib/Rekuho.svelte:108]: I feel like we at some point want to have some better separation  for this sort of logic to stay somewhere else, and to have positional comparison be somewhere else, too, but for now this is fine!

**Review by elkowar — COMMENTED** [review]
The JS part of this looks good, @buffet is gonna review the rust part

File: node_packages/bazed-svelte/src/lib/Rekuho.svelte (hunk 1 above)
Comment (druskus20) [reply node_packages/bazed-svelte/src/lib/Rekuho.svelte:108]: yep, agreed

**Review by gwimm — DISMISSED** [review]
Rounded corners are a must, I'm sure people are going to stick with VSCode (c) Microsoft, due to Bazed lacking rounded corners.

File: crates/bazed-core/src/buffer.rs [hunk 2]
```diff
@@ -85,6 +85,20 @@ impl Buffer {
                 .expect("Caret stored in BufferRegions was not a valid offset into the buffer")
         })
     }
+
+    // Great naming... better than all_caret_coordinate_regions
```
Comment (buffet) [inline crates/bazed-core/src/buffer.rs:89]: I'm not sure this is better, also I'm not sure that's the proper place to make fun of elk.

File: crates/bazed-rpc/src/core_proto.rs [hunk 3]
```diff
@@ -9,21 +9,31 @@ use uuid::Uuid;
 #[serde(transparent)]
 pub struct RequestId(pub Uuid);
 
-/// Absolute position within a document
+/// Absolute position within a document.
+/// It is called "Coordinate" instead of position because
+/// PositionRegion is not a good name.
```
Comment (buffet) [inline crates/bazed-rpc/src/core_proto.rs:14]: This is really not what doc comments are for.

**Review by buffet — CHANGES_REQUESTED** [review]
Fix commit or SASS shall stay! 🗡️

File: crates/bazed-core/src/buffer.rs (hunk 2 above)
Comment (elkowar) [reply crates/bazed-core/src/buffer.rs:89]: Note: that was not the previous name. there was no previous name.

File: node_packages/bazed-svelte/src/lib/Rekuho.svelte (hunk 1 above)
Comment (buffet) [reply node_packages/bazed-svelte/src/lib/Rekuho.svelte:108]: So true!

File: crates/bazed-core/src/buffer.rs (hunk 2 above)
Comment (druskus20) [reply crates/bazed-core/src/buffer.rs]: fine

File: node_packages/bazed-svelte/src/lib/Rekuho.svelte (hunk 1 above)
Comment (druskus20) [reply node_packages/bazed-svelte/src/lib/Rekuho.svelte:108]: don't reopen the conversation to shitpost

Comment (buffet) [reply node_packages/bazed-svelte/src/lib/Rekuho.svelte:108]: Hey, I was playing around with plugins, not shitposting. :(

**Review by buffet — APPROVED** [review]
Looks much cooler now.

**Comment by druskus20** [comment]
No you

**Comment by druskus20** [comment]
Bye sass @gwimm

**Pull request merged**